digicampipe/io/event_stream.py

```diff
@@ -42,18 +42,20 @@ def event_stream(filelist, source=None, max_events=None, **kwargs):
     else:
 
         file_stream = tqdm(filelist, total=n_files, desc='Files', leave=False)
-
     for file in file_stream:
         if source is None:
             source = guess_source_from_path(file)
         data_stream = source(url=file, **kwargs)
-        for event in data_stream:
-
-            if count >= max_events:
-                raise StopIteration
-
-            count += 1
-            yield event
+        try:
+            for event in data_stream:
+                if count >= max_events:
+                    raise StopIteration
+                count += 1
+                yield event
+        except EOFError as e:
+            print('WARNING: unexpected end of file.', e)
```

Comment:
Instead of print("Warning: ... ") one could use [the warnings package](https://docs.python.org/3.6/library/warnings.html)

Reply:
good point, fixed in next push